Products/ZenModel/Device.py

```diff
@@ -400,12 +400,36 @@ def getRRDTemplates(self):
         """
         if not hasattr(self, 'zDeviceTemplates'):
             return ManagedEntity.getRRDTemplates(self)
-        result = []
-        for name in self.zDeviceTemplates:
-            template = self.getRRDTemplateByName(name)
+        templates = []
+        for templateName in self.zDeviceTemplates:
+            if templateName.endswith('-replacement') or \
+                    templateName.endswith('-addition'):
+                continue
+
+            template = self.getRRDTemplateByName(templateName)
+            replacement = self.getRRDTemplateByName(
+                '{}-replacement'.format(templateName))
+
+            if replacement and replacement not in templates:
+                templates.append(replacement)
+                #self.setZenProperty(
+                #    'zDeviceTemplates',
+                #    self.zDeviceTemplates + [replacement.titleOrId()])
+            else:
+                templates.append(template)
+
+            addition = self.getRRDTemplateByName(
+                '{}-addition'.format(templateName))
+
+            if addition and addition not in templates:
+                templates.append(addition)
+                #self.setZenProperty(
+                #    'zDeviceTemplates',
+                #    self.zDeviceTemplates + [addition.titleOrId()])
+
             if template:
-                result.append(template)
-        return result
+                templates.append(template)
```

Comment:
This code block logic is replaced by 418-419, 'Only add if -replacement named template does not exist' ?